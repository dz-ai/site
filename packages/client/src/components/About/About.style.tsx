import styled from 'styled-components/macro'

export const ColumnWrapper = styled.div`
  @media (min-width: 768px) {
    column-gap: 5rem;
    column-count: 2;
  }
`
