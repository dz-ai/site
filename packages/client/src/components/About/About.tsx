import React from 'react'
import ColumnWrapper from '../ColumnWrapper'
import Link from '../Link/Link'
import SectionHeading from '../SectionHeading'

const About: React.FC = () => (
  <>
    <SectionHeading>About Paderborn.js</SectionHeading>
    <ColumnWrapper>
      <p>
        Paderborn.js is a usergroup focused on JavaScript and related topics. We
        meet regularly at <a href="https://enpit.de/" rel="noopener">Enpit</a>, Marienplatz 11a in Paderborn.
      </p>
      <p>
        We welcome a diverse range of topics that are related in any way to
        JavaScript and the community around it. New speakers, experienced
        speakers, and everyone in between. If you have an idea you arenâ€™t sure
        about, just reach out or add it as an issue in our Paderborn.js{' '}
        <Link href="https://github.com/paderbornjs/talks">
          talks repository
        </Link>
        !
      </p>
      <p>
        Join the{' '}
        <Link href="https://meetup.com/paderborn-js">Meetup community</Link> to
        get in contact with both organizers and other members of the community!
        Also, feel free to{' '}
        <Link href="https://twitter.com/paderbornjs">follow us on Twitter</Link>{' '}
        or tweet about our events using the hashtag <span>#paderbornjs</span>!
      </p>
      <p>
        This meetup is primarily in german, but if we have any non-german
        speaking attendees, we try to switch to english. We look forward to
        hearing from you, and to seeing you at an upcoming event!
      </p>
    </ColumnWrapper>
  </>
)

export default About
