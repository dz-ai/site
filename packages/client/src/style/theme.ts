const grey = {
  greyXDark: 'hsl(199, 40%, 28%)',
  greyDark: 'hsl(199, 38%, 47%)',
  greyMDark: 'hsl(199, 36%, 66%)',
  greyMedium: 'hsl(199, 24%, 85%)',
  greyMBright: 'hsl(199, 30%, 94%)',
  greyBright: 'hsl(199, 38%, 98%)',
  greyXBright: 'hsl(199, 40%, 100%)',
}

const primary = {
  primaryXDark: 'hsl(219, 100%, 20%)',
  primaryDark: 'hsl(219, 90%, 30%)',
  primaryMDark: 'hsl(219, 80%, 40%)',
  primaryMedium: 'hsl(219, 70%, 50%)',
  primaryMBright: 'hsl(219, 80%, 68%)',
  primaryBright: 'hsl(219, 90%, 85%)',
  primaryXBright: 'hsl(219, 100%, 93%)',
}

const accent = {
  accentXDark: 'hsl(54, 100%, 22%)',
  accentDark: 'hsl(54, 97%, 28%)',
  accentMDark: 'hsl(54, 95%, 41%)',
  accentMedium: 'hsl(54, 93%, 54%)',
  accentMBright: 'hsl(54, 95%, 65%)',
  accentBright: 'hsl(54, 97%, 76%)',
  accentXBright: 'hsl(54, 100%, 87%)',
}

export default {
  breakpoints: ['550px', '768px'],
  fontSizes: [12, 14, 16, 18, 20, 24, 30, 36, 48, 60, 72],
  space: [
    0,
    4,
    8,
    12,
    16,
    24,
    32,
    48,
    64,
    96,
    128,
    192,
    256,
    384,
    512,
    640,
    768,
  ],
  fonts: {
    sans:
      '-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI",Roboto,"Noto Sans",Ubuntu,Cantarell,"Helvetica Neue",Arial,sans-serif',
    serif: 'Merriweather,serif',
  },
  colors: {
    ...grey,
    ...primary,
    ...accent,
  },
}
