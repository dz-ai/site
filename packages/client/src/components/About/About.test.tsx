import React from 'react'
import { render } from 'react-testing-library'
import { ThemeProvider } from 'styled-components/macro'
import theme from '../../style/theme'
import About from './About'

describe('<About />', () => {
  test('matches snapshot', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <About />
      </ThemeProvider>
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
