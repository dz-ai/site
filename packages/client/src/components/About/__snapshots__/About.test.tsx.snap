// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> matches snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  margin-top: 0;
  margin-bottom: 16px;
  font-size: 20px;
  font-family: Merriweather,serif;
  font-weight: 300;
  text-align: center;
}

@media screen and (min-width:550px) {
  .c0 {
    margin-bottom: 24px;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    margin-bottom: 32px;
  }
}

@media screen and (min-width:550px) {
  .c0 {
    font-size: 30px;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    font-size: 30px;
  }
}

<h2
  class="c0"
  font-family="serif"
  font-size="4,6,6"
  font-weight="300"
>
  About Paderborn.js
</h2>
`;
